packages/hash/frontend/src/pages/[account-slug]/types/entity-type/custom-property-type-menu.tsx

```diff
@@ -0,0 +1,221 @@
+import { faClose, faList } from "@fortawesome/free-solid-svg-icons";
+import {
+  Button,
+  Chip,
+  ChipProps,
+  FontAwesomeIcon,
+} from "@hashintel/hash-design-system";
+import {
+  Box,
+  buttonClasses,
+  chipClasses,
+  Stack,
+  Typography,
+} from "@mui/material";
+import { uniqueId } from "lodash";
+import { FunctionComponent } from "react";
+import { useFormContext, useWatch } from "react-hook-form";
+import { faCube } from "../../../../shared/icons/pro/fa-cube";
+import { ArrayPropertyTypeMenu } from "./array-property-type-menu";
+import { PropertyTypeFormValues } from "./property-type-form";
+import { usePropertyTypeSelectorDropdownContext } from "./property-type-selector-dropdown";
+import { getDefaultData } from "./property-type-utils";
+
+const CustomChip: FunctionComponent<ChipProps & { borderColor?: string }> = ({
+  borderColor,
+  ...props
+}) => (
+  <Chip
+    {...props}
+    sx={{
+      borderColor: borderColor ?? "transparent",
+      [`.${chipClasses.label}`]: { paddingX: 1, paddingY: 0.25, fontSize: 11 },
+    }}
+  />
+);
+
+type CustomPropertyTypeMenuProps = {
+  closeMenu: () => void;
+};
+
+export const CustomPropertyTypeMenu: FunctionComponent<
+  CustomPropertyTypeMenuProps
+> = ({ closeMenu }) => {
+  const { closeCustomPropertyMenu } = usePropertyTypeSelectorDropdownContext();
+
+  const { getValues, setValue, control } =
+    useFormContext<PropertyTypeFormValues>();
+  const creatingPropertyId = useWatch({ control, name: "creatingPropertyId" });
+
+  return (
+    <Box>
+      <Stack
+        sx={({ palette }) => ({
+          background: palette.gray[10],
+          border: `1px solid ${palette.gray[30]}`,
+          borderTopRightRadius: 4,
+          borderTopLeftRadius: 4,
+          paddingX: 2.75,
+          paddingY: 2,
+          borderBottomWidth: 0,
+        })}
+      >
+        <Stack direction="row" justifyContent="space-between">
+          <Typography
+            variant="smallCaps"
+            sx={{ color: ({ palette }) => palette.gray[70] }}
+          >
+            Specify a custom expected value
+          </Typography>
+
+          <Button
+            onClick={closeMenu}
+            size="small"
+            sx={{
+              padding: 0,
+              minHeight: 0,
+              background: "none !important",
+              fontWeight: 600,
+              fontSize: 12,
+              color: ({ palette }) => palette.gray[50],
+              [`.${buttonClasses.endIcon}`]: {
+                ml: 0,
+              },
+            }}
+            endIcon={<FontAwesomeIcon icon={faClose} />}
+            variant="tertiary_quiet"
+          >
+            CANCEL
```

Comment:
Can we put a bit of margin to the left of the CANCEL text? It looks a bit crowded at the moment

**Current**
![Screenshot 2022-12-02 at 08 44 22](https://user-images.githubusercontent.com/37743469/205252454-2c831782-cd9a-42ce-b025-eefda3f87005.png)

**Design**
![Screenshot 2022-12-02 at 08 44 15](https://user-images.githubusercontent.com/37743469/205252484-4d554c02-ed9e-4ed8-aac2-c355df5c3a5a.png)

Reply:
Made several changes to this button in dd9f15bc762278ef8bad137dd33f88cfc2997dc6. Thank you.